src/VscProcess.cpp

```diff
@@ -457,21 +481,25 @@ void VscProcess::readFromVehicle()
 
   // Log warning when no data is received
   ros::Duration noDataDuration = ros::Time::now() - lastDataRx;
-  if (noDataDuration > ros::Duration(.25))
+  ros::Duration reconnectTimeout = ros::Time::now() - lastReconnectAttempt;
+  if (noDataDuration > ros::Duration(.25) && reconnectTimeout > ros::Duration(reconnect_time_))
   {
     ROS_WARN_THROTTLE(.5, "No Data Received in %i.%09i seconds", noDataDuration.sec, noDataDuration.nsec);
     ROS_WARN_STREAM("Attempting to reconnect to the VSC...");
+    
+    lastReconnectAttempt = ros::Time::now();
```

Comment:
seeing `ros::Time::now()` called many times.
Can you please make a new variable that saves that after the `// Log warning when no data is received` comment

Reply:
Made method-scope variable for current time.